components/script/dom/webglrenderingcontext.rs

```diff
@@ -227,18 +274,36 @@ impl WebGLRenderingContext {
         &self.limits
     }
 
-    pub fn bound_texture_for_target(&self, target: &TexImageTarget) -> Option<DomRoot<WebGLTexture>> {
-        match *target {
-            TexImageTarget::Texture2D => self.bound_texture_2d.get(),
-            TexImageTarget::CubeMapPositiveX |
-            TexImageTarget::CubeMapNegativeX |
-            TexImageTarget::CubeMapPositiveY |
-            TexImageTarget::CubeMapNegativeY |
-            TexImageTarget::CubeMapPositiveZ |
-            TexImageTarget::CubeMapNegativeZ => self.bound_texture_cube_map.get(),
+    fn bound_texture(&self, target: u32) -> Option<DomRoot<WebGLTexture>> {
```

Comment:
If we take a custom enum argument that only accepts Texture2d and TextureCubeMap, we don't need to return an Option. This will simplify other calling code, and then we can remove the optional_root_object_tojs_or_null macro.

Reply:
Ok, I'll create the new enum. The result will be an Option too because the bound texture is nullable.

Reply:
Oh, I see. In that case I don't think there's any strong benefit to using a separate enum.